Tests/SonarScanner.MSBuild.PreProcessor.Test/PreprocessorObjectFactoryTests.cs

```diff
@@ -39,21 +43,57 @@ public class PreprocessorObjectFactoryTests
             logger = new TestLogger();
 
         [TestMethod]
-        public void Factory_ThrowsOnInvalidInput()
+        public void CreateSonarWebService_ThrowsOnInvalidInput()
         {
             ((Func<PreprocessorObjectFactory>)(() => new PreprocessorObjectFactory(null))).Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("logger");
 
             var sut = new PreprocessorObjectFactory(logger);
-            sut.Invoking(x => x.CreateSonarWebService(null)).Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("args");
+            sut.Invoking(x => x.CreateSonarWebService(null).Result).Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("args");
+        }
+
+        [DataTestMethod]
+        [DataRow("https://sonarsource.com/", "https://sonarsource.com/api/server/version")]
+        [DataRow("https://sonarsource.com", "https://sonarsource.com/api/server/version")]
+        [DataRow("https://sonarsource.com/sonarlint", "https://sonarsource.com/sonarlint/api/server/version")]
+        [DataRow("https://sonarsource.com/sonarlint/", "https://sonarsource.com/sonarlint/api/server/version")]
+        public async Task CreateSonarWebService_AppendSlashSuffixWhenMissing(string input, string expected)
+        {
+            var sut = new PreprocessorObjectFactory(logger);
+
+            var downloader = new Mock<IDownloader>();
+            downloader.Setup(x => x.Download(It.Is<Uri>(uri => uri.ToString() == expected), false))
+                .Returns(Task.FromResult("9.9"))
+                .Verifiable();
+
+            var validArgs = CreateValidArguments(input);
+            await sut.CreateSonarWebService(validArgs, downloader.Object);
+
+            downloader.VerifyAll();
```

Comment:
This will verify that the setup method was actually executed with the correct parameter.